components/ProgressBar.qml

```diff
@@ -44,24 +45,19 @@ Rectangle {
         }
 
         if(targetBlock > 0) {
-            var remaining = targetBlock - currentBlock
-            // wallet sync
-            if(blocksToSync > 0)
-                var progressLevel = (100*(blocksToSync - remaining)/blocksToSync).toFixed(0);
-            // Daemon sync
-            else
-                var progressLevel = (100*(currentBlock/targetBlock)).toFixed(0);
+            var remaining = (currentBlock < targetBlock) ? targetBlock - currentBlock : 0
+            var progressLevel = (blocksToSync > 0) ? (100*(blocksToSync - remaining)/blocksToSync).toFixed(0) : 100
```

Comment:
In these two lines, the difference between targetBlock and blocksToSync isnt't clear. For the progress level to match remaining, they shoiuld be the same, no (plus or minus one) ?

Reply:
It's related to #387. If the difference between currentBlock and targetBlock is small, the progress bar would be close to 100% filled during the whole sync progress. 

blocksToSync is set at wallet startup and the progress bar indicated the sync progress from 0% to 100% in the current session. If you restart the wallet, the bar restarts from 0%.

Reply:
OK, that makes sense.

Reply:
I have a bug where the daemon progress is always 100%. Possibly happens when `blocksToSync` is undefined, see how its used in `main.qml`.